### DISCUSSION TITLE
Extending Generated Resolvers With Relations

### DISCUSSION THREAD

**Discussion by tradical-allstripes** [MichalLytek/typegraphql-prisma#104] [General]
Hello! I apologize if this is documented somewhere, but I couldn't find it.

I'm currently trying to add a field resolver to one of the types generated by this plugin, and running into some trouble. Specifically, the relations do not resolve properly. Here is my first attempt with any confidential information removed:

```typescript
@Resolver(() => Foo)
export class CustomFooResolver extends FooCrudResolver {
  @FieldResolver(() => Date)
  derivedField(@Root() foo: Foo) {
    return foo?.barObj?.baz
  }
}
```

The issue I'm seeing is that `barObj` is undefined. I have also confirmed that **none** of the relation fields are populated. We also have a separate field for the `bar` string id, which is being populated as expected. Note that I am including the `FooRelationsResolver` in the schema registry, as the documentation says. Relations resolve as expected using the raw generated resolver. I've tried adding the following (and registering it) with no change in behavior:

```typescript
@Resolver(() => Foo)
export class CustomFooRelationsResolver extends FooRelationsResolver {
}
```

What am I doing wrong? Is what I'm trying to do even possible?

Thanks!

**Comment by MichalLytek** [comment]
In TypeGraphQL (and GraphQL implementations at all) you never can't access the data resolved in other field resolved.
Data returned from field resolvers are never attached to the root object. They are a virtual field and will be returned only in final graphql response.
So for `typegraphql-prisma` all models relations are implemented as field resolvers so you can access `barObj`. 

I probably should remove `barObj` from the types signature at all. It was mostly designed to be compatible with prisma return types, but TS allows for additional properties in returned objects so it doesn't matter.

**Reply by tradical-allstripes** [reply]
Thanks as always for the refreshingly prompt reply!

Anyway, what you're saying makes sense and is good information to have. Do you happen to have an idea off hand of how/if these custom field resolvers like I'm trying to do could be achieved without making a separate prisma query?

**Reply by MichalLytek** [reply]
You would need a caching layer, so getting the same data from prisma again won't ends up in a sql query.

Field resolvers in graphql and independent and can be called in any order, so you can't rely on assigning data back to `root` object.

**Reply by tradical-allstripes** [reply]
Ok, thank you!
